src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,60 @@
+package duke.parser;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class Parser {
+
+    public String parseCommand(String s) {
+        if (s.toLowerCase().equals("bye")) {
+            return "bye";
+        } else if (s.toLowerCase().equals("list")) {
+            return "list";
+        } else if (s.toLowerCase().startsWith("todo")) {
+            return "todo";
+        } else if (s.toLowerCase().startsWith("done")) {
+            return "done";
+        } else if (s.toLowerCase().startsWith("find")) {
+            return "find";
+        } else if (s.toLowerCase().startsWith("event")) {
+            return "event";
+        } else if (s.toLowerCase().startsWith("delete")) {
+            return "delete";
+        } else if (s.toLowerCase().startsWith("deadline")) {
+            return "deadline";
+        } else {
+            return "unknown command detected";
+        }
+    }
+
+    public String parseDesc(String s) {
+        String cmd = parseCommand(s);
+        int index = 0;
+        switch (cmd) {
+            case "deadline":
+                index = s.indexOf("/");
+                return s.substring(9, index - 1).trim();
+            case "event":
+                index = s.indexOf("/");
+                return s.substring(6, index - 1).trim();
+            case "todo":
+            case "find":
+                return s.substring(5).trim();
+            default:
+                return "No description provided";
+        }
+    }
+
+    public String parseDate(String s) {
+        int index = s.indexOf("/");
+        String rawDate = s.substring(index + 4).trim();
+        SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy HHmm");
+        try {
+            Date d = sdf.parse(rawDate);
+            return d.toString();
+        } catch (ParseException e) {
+            return rawDate;
```

Comment:
Perhaps could help the user enter the correct format by validating and displaying an error in Ui instead

This current variation allows the following variant to be added to the tasklist
event hello /s werh

Program displays:

Got it. I've added this task:
[E][✘] hello (at: erh)
Now you have 5 tasks in the list.